proto/ignition/msgs/particle_emiter.proto

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+syntax = "proto3";
+package ignition.msgs;
+option java_package = "com.ignition.msgs";
+option java_outer_classname = "ParticleEmitterProtos";
+
+/// \ingroup ignition.msgs
+/// \interface ParticleEmitter
+/// \brief Message for a particle emitter.
+
+import "ignition/msgs/color.proto";
+import "ignition/msgs/header.proto";
+import "ignition/msgs/material.proto";
+import "ignition/msgs/pose.proto";
+import "ignition/msgs/vector3d.proto";
+
+message ParticleEmitter
+{
+  /// \brief Optional header data.
+  Header header                 = 1;
+
+  /// \brief The emitter name.
+  string name                   = 2;
+
+  /// \brief All possible emitter types.
+  enum EmitterType
+  {
+    /// \brief Point emitter.
+    POINT     = 0;
+    /// \brief Box emitter.
+    BOX       = 1;
+    /// \brief Cylinder emitter.
+    CYLINDER  = 2;
+    /// \brief Ellipsoid emitter.
+    ELLIPSOID = 3;
+  }
+  /// \brief The emitter type.
+  EmitterType type              = 3;
+
+  /// \brief The position of the emitter.
+  Pose pose                     = 4;
+
+  /// The size of the emitter where the particles are sampled.
+  Vector3d size                 = 5;
+
+  /// \brief How many particles per second should be emitted.
+  float rate                    = 6;
+
+  /// \brief The number of seconds the emitter is active.
+  float duration                = 7;
+
+  /// \brief Whether particle emitter is enabled or not.
+  bool emitting                 = 8;
+
+  /// \brief The particle dimensions (width, height, depth).
+  Vector3d particle_size        = 9;
+
+  /// \brief The number of seconds each particle will ’live’ for before
+  /// being destroyed.
+  float lifetime                = 10;
```

Comment:
Just curious, how does this interact with `duration`?

Reply:
The `lifetime` parameter sets the life of a single particle. Particles will disappear and new particles will be emitted. This cycle will repeat for a number of seconds set by the `duration` parameter. In other words, `duration` is a timer for the general switch.